### ISSUE TITLE
p-menu: command function losing context

### ISSUE THREAD

**Issue by PaulSchult** [primefaces/primeng#13776]
### Describe the bug

When using a p-menu that uses a command function, as seen in the stackblitz, the value that was once assigned doesn't get updated after the first click. This worked with Angular 14. 

Working example with Angular 14. [see here](https://stackblitz.com/edit/stackblitz-starters-u5ny8h?file=README.md,src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts,src%2Findex.html,src%2Fapp%2Fapp.module.ts,src%2Fapp%2Fapp.component.spec.ts)

### Environment

See Stackblitz link.

### Reproducer

https://stackblitz.com/edit/cp3ztu?file=src%2Fapp%2Fdemo%2Fmenu-basic-demo.html,src%2Fapp%2Fdemo%2Fmenu-basic-demo.ts

### Angular version

16.2.0

### PrimeNG version

16.3.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

-

### Steps to reproduce the behavior

1. Open Stackblitz link
2. Click "Deutsch" and see value "de" on the right
3. Click "English" and see that the value hasn't changed
4. Refresh the page and this time select "English" first
5. Notifce that "en" is displayed on the right
6. Click "Deutscht" and see that the value hasn't changed

### Expected behavior

The value should change depending on which item I'm selecting.

**Comment by ashikjs** [comment]
```
 for (const language of languages) {
      this.items.push({
        label: language.translations.de,
        command: () => {
          console.log(language.value);
          this.currentLanguage = language.value;
        },
      });
    }
```

this is workable version of your code. 

you can check it here: https://stackblitz.com/edit/cp3ztu-tykkvq?file=src%2Fapp%2Fdemo%2Fmenu-basic-demo.html,src%2Fapp%2Fdemo%2Fmenu-basic-demo.ts

**Comment by PaulSchult** [comment]
> ```
>  for (const language of languages) {
>       this.items.push({
>         label: language.translations.de,
>         command: () => {
>           console.log(language.value);
>           this.currentLanguage = language.value;
>         },
>       });
>     }
> ```
> 
> this is workable version of your code.
> 
> you can check it here: https://stackblitz.com/edit/cp3ztu-tykkvq?file=src%2Fapp%2Fdemo%2Fmenu-basic-demo.html,src%2Fapp%2Fdemo%2Fmenu-basic-demo.ts

sorry, I should have clarified that the async is important. I am doing some async calculations inside the command function

**Comment by ashikjs** [comment]
> > ```
> >  for (const language of languages) {
> >       this.items.push({
> >         label: language.translations.de,
> >         command: () => {
> >           console.log(language.value);
> >           this.currentLanguage = language.value;
> >         },
> >       });
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > this is workable version of your code.
> > you can check it here: https://stackblitz.com/edit/cp3ztu-tykkvq?file=src%2Fapp%2Fdemo%2Fmenu-basic-demo.html,src%2Fapp%2Fdemo%2Fmenu-basic-demo.ts
> 
> sorry, I should have clarified that the async is important. I am doing some async calculations inside the command function

Can you add some real life sync example here so we can understand.

**Comment by PaulSchult** [comment]
> Can you add some real life sync example here so we can understand.

Do you mean async? E.g we have a dropdown where you can select a template and on the right is the menu button where the user can select the language of the template. Inside of the command function we take the selected template and load it from the db with the given language and apply it to some editor. The fetching is async, thats why we need the function to be async.

**Comment by cetincakiroglu** [comment]
Hi,

The issue is not exist in the latest PrimeNG, it might fixed in the versions between 16.2.x-16.4.x

See the working example here: https://stackblitz.com/edit/zrdb2w?file=src%2Fapp%2Fdemo%2Fmenu-command-demo.ts,src%2Fapp%2Fdemo%2Fmenu-command-demo.html
